java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java

```diff
@@ -1146,6 +1147,52 @@ public void infer_fully_lambda_types() {
     JavaType Rsubstitution = typeSubstitution.substitutedTypes().get(1);
     assertThat(Rsubstitution.is("java.lang.String[]")).isTrue();
 
+    lambda = ((LambdaExpressionTree) result.symbol("line1").declaration().parent());
```

Comment:
The checks on `line1`, `line2`, and `line` could go into separate methods, that would increase readability.